x-pack/legacy/plugins/infra/public/pages/logs/analysis/sections/log_rate.tsx

```diff
@@ -0,0 +1,230 @@
+
+const areaSeriesColour = 'rgb(224, 237, 255)';
+const lineSeriesColour = 'rgb(49, 133, 252)';
+
+export const LogRateResults = ({
+  isLoading,
+  results,
+}: {
+  isLoading: boolean;
+  results: GetLogEntryRateSuccessResponsePayload['data'] | null;
+}) => {
+  const title = i18n.translate('xpack.infra.logs.analysis.logRateSectionTitle', {
+    defaultMessage: 'Log entries',
+  });
+
+  const loadingAriaLabel = i18n.translate(
+    'xpack.infra.logs.analysis.logRateSectionLoadingAriaLabel',
+    { defaultMessage: 'Loading log rate results' }
+  );
+  const showModelBoundsLabel = i18n.translate(
+    'xpack.infra.logs.analysis.logRateSectionModelBoundsCheckboxLabel',
+    { defaultMessage: 'Show model bounds' }
+  );
+
+  const { areaSeries, lineSeries, anomalySeries } = useLogEntryRateGraphData({
+    data: results,
+  });
+
+  const dateFormatter = useMemo(
+    () =>
+      lineSeries.length > 0
+        ? niceTimeFormatter([first(lineSeries)[0], last(lineSeries)[0]])
+        : (value: number) => `${value}`,
+    [lineSeries]
+  );
+
+  const areaSpecId = getSpecId('modelBounds');
+  const lineSpecId = getSpecId('averageValues');
+  const anomalySpecId = getSpecId('anomalies');
+
+  const [dateFormat] = useKibanaUiSetting('dateFormat');
+
+  const tooltipProps = {
+    headerFormatter: useCallback(
+      (data: TooltipValue) => moment(data.value).format(dateFormat || 'Y-MM-DD HH:mm:ss.SSS'),
+      [dateFormat]
+    ),
+  };
+
+  const [isShowingModelBounds, setIsShowingModelBounds] = useState<boolean>(true);
+
+  return (
+    <>
+      <EuiTitle size="m" aria-label={title}>
+        <h2>{title}</h2>
+      </EuiTitle>
+      <EuiSpacer size="l" />
+      {isLoading ? (
+        <EuiFlexGroup justifyContent="center">
+          <EuiFlexItem grow={false}>
+            <EuiLoadingChart size="xl" aria-label={loadingAriaLabel} />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      ) : !results || (results && results.histogramBuckets && !results.histogramBuckets.length) ? (
+        <EuiEmptyPrompt
+          title={
+            <h2>
+              {i18n.translate('xpack.infra.logs.analysis.logRateSectionNoDataTitle', {
+                defaultMessage: 'There is no data to display.',
+              })}
+            </h2>
+          }
+          titleSize="m"
+          body={
+            <p>
+              {i18n.translate('xpack.infra.logs.analysis.logRateSectionNoDataBody', {
+                defaultMessage: 'Try adjusting your time range',
+              })}
+            </p>
+          }
+        />
+      ) : (
+        <>
+          <EuiFlexGroup justifyContent="flexEnd">
+            <EuiFlexItem grow={false}>
+              <EuiCheckbox
+                id={'showModelBoundsCheckbox'}
+                label={showModelBoundsLabel}
+                aria-label={showModelBoundsLabel}
+                checked={isShowingModelBounds}
+                onChange={e => {
+                  setIsShowingModelBounds(e.target.checked);
```

Comment:
Hmm, yeah, good point / good catch. 

So my feelings are we shouldn't keep the y-axis scale the same with model bounds on and off - whilst it would prevent the "jumping" on the initial switch, the scales between the data points of the log entries and the model bounds have the potential to be quite huge, and this would lead to odd looking graphs where the top and bottom aren't rendering any data points.

I wondered if this was just because I wasn't feeding the Y axis component a `domain` prop (an optional prop), by default my expectation is that graphs will use the `min` and `max` of all of the available `yAccessors` data points to decide the scale / domain for the axis, but this doesn't seem to be the case here.

I've just tried out using the `domain` prop and it didn't make a difference, so I'm wondering if this is an elastic-charts bug?

For context I added:

```
const yAxisDomain = useMemo(() => {
    const yDataPoints: number[] = [];
    if (isShowingModelBounds) {
      areaSeries.forEach(dataPoint => {
        dataPoint.min && yDataPoints.push(dataPoint.min);
        dataPoint.max && yDataPoints.push(dataPoint.max);
      });
    }
    lineSeries.forEach(dataPoint => {
      dataPoint[1] && yDataPoints.push(dataPoint[1]);
    });
    anomalySeries.forEach(dataPoint => {
      yDataPoints.push(dataPoint[1]);
    });
    return { min: min(yDataPoints), max: max(yDataPoints) };
  }, [areaSeries, lineSeries, anomalySeries, isShowingModelBounds]);
```

then added the prop:

```
<Axis
                id={getAxisId('values')}
                title={i18n.translate('xpack.infra.logs.analysis.logRateSectionYaxisTitle', {
                  defaultMessage: 'Log entries',
                })}
                position="left"
                tickFormat={value => Number(value).toFixed(0)}
                domain={yAxisDomain}
              />
```

But this still rendered:

<img width="1679" alt="Screenshot 2019-08-19 16 25 12" src="https://user-images.githubusercontent.com/471693/63278620-88124c80-c29f-11e9-8a3f-5424d4d48d84.png">

I'm not really sure why the axis is maxing out at `22000` when the `max` on the domain is `23178`.

I don't think padding would make a difference here, as the scale is just incorrect 😬

Reply:
Yeah I didn't mean CSS padding, I meant scale padding (react-viz had some concept of the domain having "padding" I think, but no idea if elastic charts has anything like that). Probably another thing to check in with them on about why this is rendering the way it is ... possibly not taking into account the anomalies series for the domain calculation for some reason?

Reply:
@jasonrhodes Yeah, all good, I knew you meant padding on the scale, I was just tying to say that if the underlying values are wrong anyway, padding doesn't help.

> possibly not taking into account the anomalies series for the domain calculation for some reason?

I did think this, which is why I tried forcibly providing the `domain` prop (which uses values from all 3, anomalies included), but it came out with the same scale. I'll enquire with the experts 😄

Reply:
Followup: fixes for this are in progress on the charts side 👍